Chromap read format

mei-knudson opened the pull request:
Submitting this as a draft

Removing correcting and trimming
Removing "no_align" pipeline mode
Adding get_read_format task prior to Chromap

Not sure if we need to add any additional logic
Not sure if it would make more sense to have the read format task input be fastq_barcode[0] instead of read2[0]
Would appreciate feedback on task name, variable names, structure or anything else

emattei commented on tasks/task_chromap_read_format.wdl:
```diff
@@ -0,0 +1,53 @@
+
+        echo "r1:0:-1,r2:0:${r2_end},bc:${bc1_start}:${bc1_end},bc:${bc2_start}:${bc2_end},bc:${bc3_start}:${bc3_end}" > read_format.txt
+
+        cat read_format.txt
```
Instead of this you can use `tee` in the previous command
`...${bc3_end}" | tee -a read_format.txt`

double check the syntax

emattei commented on tasks/task_chromap_read_format.wdl:
```diff
@@ -0,0 +1,53 @@
+# TASK
+# SHARE-chromap-read-format
+
+task get_read_format {
```
I would add `chromap` somewhere because this works only for chromap

emattei commented on workflows/subwf-atac.wdl:
```diff
@@ -1,8 +1,7 @@
 # Import the tasks called by the pipeline
-import "../tasks/share_task_correct_fastq.wdl" as task_correct_fastq
-import "../tasks/task_trim_fastqs_atac.wdl" as task_trim
+import "../tasks/task_chromap_read_format.wdl" as task_chromap_read_format
```
oh you do have chromap here. so call it get_chromap_read_format and change it in the WDL as well

emattei commented on workflows/subwf-atac.wdl:
```diff
@@ -101,47 +83,18 @@ workflow wf_atac {
+    if ( "~{chemistry}" == "shareseq" ) {
+        call task_chromap_read_format.get_read_format as get_read_format {
+            input:
+                fastq_R2 = read2[0]
```
it should just be fastq the input. and yes you should pass fastq_barcode

emattei commented on workflows/subwf-atac.wdl:
```diff
@@ -164,70 +117,63 @@ workflow wf_atac {
             bc_error_threshold = bc_error_threshold,
             bc_probability_threshold = bc_probability_threshold,
-            read_format = read_format
+            read_format = select_first([get_read_format.read_format, read_format])
```
This might create problems in the future because we will never be able to use our own  read_format for share.
If we change something to the files in the future we will have to change the task to be able to test. Also, do we need still need linkers since we removed the shifting?

mei-knudson replied:
What do you mean by "do we still need linkers"? You mean like to remove the linker sequences when writing the barcodes into the fastq files during demux?

emattei reviewed (changes requested):
why STARsolo doesn't need to have the parameters updated?

mei-knudson commented:
> why STARsolo doesn't need to have the parameters updated?

You mean for specifying where the barcodes are? For STARsolo, I used negative indexing to specify the barcode positions; since it's indexing from the end, the R2 read length doesn't matter. But it makes sense that we might want to pass in our own read_format, I'll try to make the STARsolo task accommodate that

mei-knudson commented:
Made suggested changes (tee, get_chromap_read_format, fastq = fastq_barcode[0]).

I changed the logic in subwf_atac so that task_chromap_read_format is called if chemistry=='shareseq' AND read_format is not defined, so that we can pass in custom a read_format for SHARE. I set the default value of read_format to "bc:0:-1,r1:0:-1,r2:0:-1" in the chromap task (would be used for 10X).

I added a soloCBposition input for STARsolo and set the default value to be the string we use for specifying the SHARE barcodes. The default should be ok since we don't use soloCBposition when we run STARsolo on 10X. 

Tested both chromap and STARsolo, ran successfully

emattei reviewed (approved):
I am merging this but it would be nice to have an equivalent task get read format starsolo. But I am thinking about what would be the best way
